### PR TITLE
Fix/update poll item answer

### PR DESCRIPTION
**Pull request by Splines**
## 👩‍💻 New algorithm for updating the answers of a poll item
Update the answers of a poll item according to an update list of selection options (strings).

Therefore added a method that operates in-place on the poll item answers of the database and updates them according to the items from the poll. Further requirements of this algorithm are described [here](https://github.com/livepoll/live-poll-api/blob/ac80f4fa1a0e4a0dc44f27219d368dae810d91e0/src/main/kotlin/de/livepoll/api/service/PollItemService.kt#L183-L196):

```kotlin
 * Update the answers of a poll item according to an update list of selection options (strings).
 * This method works in-place and will adjust the poll item answers list.
 *
 * Requirements this algorithm has to fulfill:
 *
 * - Selection option already exists (in an answer in the database)
 * -> Keep the existing one (including the answer count)
 *
 * - Selection option is new
 * -> Add a new answer to this poll item
 *
 * - Selection option existed in the database but not anymore in the update
 * -> Remove the answer that contained this selection option
 */
fun updateAnswers(item: PollItemAnswerable, selectionOptionsUpdate: List<String>) {
```

In order to make sure that this properly works, I've added quite some tests to postman. Note that they are not working with https on localhost (see [this Postman issue](https://github.com/postmanlabs/postman-app-support/issues/8612#issuecomment-792714121)). **That's why we need to set the env variable `LIVE_POLL_HTTPS_ENABLED` to `false` for Postman/Newman tests.** @marcauberer could you please adjust this in our workflows?

## ⏫ Another inheritance layer
In order to be able to outsource the `updateAnswer` algorithm to its own function, I've added another inheritance layer: 

```
PollItem
     ^
     | 
PollItemAnswerable
This includes an answers list which is a MutableList<out SelectionOptionAnswer>.
The interface SelectionOptionAnswer is implemented by MultipleChoiceItemAnswer + QuizItemAnswer.
Therefore MultipleChoiceItem and QuizItem are allowed to extend from PollItemAnswerable.
     ^
     | 
MultipleChoiceItem + QuizItem
```

## ↔ Distinction between `answer` and `selection option`
While creating an answerable poll item (MultipleChoiceItem or QuizItem), the frontend doesn't actually pass in answers but selection options. The difference is:

- a selection option is simply a string describing an option the user can select as part of an answer
- a answer includes selection options but also an answer count

To reflect these changes, I've adjusted some DTOs. @livepoll/frontend-team : Note that these DTOs are affected:
- `MultipleChoiceItemDtoIn.kt`
- `QuizItemDtoIn.kt`

And therefore the following endpoints are affected:
- `Create multiple choice item`
- `Create quiz item`
- `Update multiple choice item`
- `Update quiz item`

### CODE REVIEW

**Review by marcauberer — APPROVED** [review]

**Comment by Splines** [comment]
Please don't squash and merge yet @marcauberer, let's wait for approval on the part of @Niklas-23 💫

File: src/main/kotlin/de/livepoll/api/entity/dto/MultipleChoiceItemDtoOut.kt
Comment (Splines) [inline src/main/kotlin/de/livepoll/api/entity/dto/MultipleChoiceItemDtoOut.kt:9]: These are supposed to be answers including selection options. Please undo this commit.

File: src/main/kotlin/de/livepoll/api/service/PollItemService.kt
Comment (Niklas-23) [inline src/main/kotlin/de/livepoll/api/service/PollItemService.kt:270]: When I look at the updateAnswer method, I think we can't be sure that the first item here is the correct one. New added items are inserted at the end of the existing list, so that the wrong item could be marked as correct (if there is a new answer at the top of the list). Take a look at my old code, that is why I am looking for the first element of the update list again

Comment (Splines) [reply src/main/kotlin/de/livepoll/api/service/PollItemService.kt]: True, thanks for noticing, I've updated the logic and added checks for whether it is allowed to have a new correct item or not.

**Review by Splines — COMMENTED** [review]
accidentally started a review....

**Comment by Splines** [comment]
@Niklas-23 please check again if the new logic is doing what you suggested

**Comment by Splines** [comment]
will add more postman tests regarding the new change later this evening

**Review by Niklas-23 — APPROVED** [review]

**Pull request merged**
